From 16f5f57a89b8c5ae67339978cdcf524b0ca74484 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:34 -0400
Subject: [block] blk-mq: fix leak of set->tags

Message-id: <1402670238-13916-47-git-send-email-snitzer@redhat.com>
Patchwork-id: 83787
O-Subject: [RHEL7.1 PATCH 046/150] blk-mq: fix leak of set->tags
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 981bd189f80f34c07ac0d87e4760d09bcda5f4cb
Author: Ming Lei <tom.leiming@gmail.com>
Date:   Thu Apr 24 00:07:34 2014 +0800

    blk-mq: fix leak of set->tags

    set->tags should be freed in blk_mq_free_tag_set().

    Signed-off-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index c26b3be..cea1bd8 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1562,6 +1562,7 @@ void blk_mq_free_tag_set(struct blk_mq_tag_set *set)
 
  for (i = 0; i < set->nr_hw_queues; i++)
   blk_mq_free_rq_map(set, set->tags[i], i);
+ kfree(set->tags);
 }
 EXPORT_SYMBOL(blk_mq_free_tag_set);
 
-- 
1.7.1